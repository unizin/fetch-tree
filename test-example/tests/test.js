/* eslint-env jest */
/* eslint-disable react/prop-types */
import React from 'react'

import { renderSnapshot } from './test-utils.js'
import Todo from '../src/containers/todo'
import TodoList from '../src/components/todo-list'
import TodoListWithPreloading from '../src/containers/todo-list-with-preloading'
import TodoWrapperWithPreloading from '../src/containers/todo-wrapper-with-preloading'


describe('single-connected-component', () => {
    test(`render a single connected component`, async () => {
        const report = await renderSnapshot(
            <Todo id={1} />
        )
        expect(report.apiRequests).toEqual({
            fetchTodo: 1,
        })
        expect(report.loadingScreens).toBe(1)
    })

    test(`render a list of connected components (no preloading)`, async () => {
        const report = await renderSnapshot(
            <TodoList ids={[1, 2, 3, 4, 5]} />
        )

        expect(report.apiRequests).toEqual({
            fetchTodo: 5,
        })
        expect(report.loadingScreens).toBe(5)
    })

    test(`preloading for a single child`, async () => {
        const report = await renderSnapshot(
            <TodoWrapperWithPreloading id={5} />
        )

        expect(report.apiRequests).toEqual({
            fetchTodo: 1,
        })
        expect(report.loadingScreens).toBe(1)
    })

    test(`preloading a list with connected children`, async () => {
        const report = await renderSnapshot(
            <TodoListWithPreloading ids={[1, 2, 3, 4, 5]} />
        )

        expect(report.apiRequests).toEqual({
            fetchTodo: 5,
        })
        expect(report.loadingScreens).toBe(1)
    })
})
