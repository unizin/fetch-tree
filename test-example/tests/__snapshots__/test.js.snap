exports[`single-connected-component preloading a list with connected children 1`] = `
<ul>
  <li>
    <div>
      id: 1
      <input
        checked={false}
        type="checkbox" />
      <p>
        Fouti hi rovumjak wuwab ojki utemun ti voblom mefup ivkiz nadahuw ci.
      </p>
    </div>
  </li>
  <li>
    <div>
      id: 2
      <input
        checked={true}
        type="checkbox" />
      <p>
        Bev rip asu rosob lo su za wirun kubvaw nadridzez cabfed temekuinu co.
      </p>
    </div>
  </li>
  <li>
    <div>
      id: 3
      <input
        checked={true}
        type="checkbox" />
      <p>
        Liuna igana pucodmek emik hapo tuczuc vokme podikru vokuse horu mum tion catke pohabe wipewen zavse osawa.
      </p>
    </div>
  </li>
  <li>
    <div>
      id: 4
      <input
        checked={false}
        type="checkbox" />
      <p>
        Mufkepgi kuretfu vucuce gardece leh nah ra gajlovzes hovwuc oca girgege zivupor ikjahwas wikesa ba ah.
      </p>
    </div>
  </li>
  <li>
    <div>
      id: 5
      <input
        checked={false}
        type="checkbox" />
      <p>
        Gem pujhi elugij mifopa bic bon ra ihaphi mit gasezuk tutfozos be iwko pulme ewehe fuene cozmo marotev.
      </p>
    </div>
  </li>
</ul>
`;

exports[`single-connected-component preloading for a single child 1`] = `
<span>
  <h1>
    Wrapper: 
    5
  </h1>
  <div>
    id: 5
    <input
      checked={false}
      type="checkbox" />
    <p>
      Gem pujhi elugij mifopa bic bon ra ihaphi mit gasezuk tutfozos be iwko pulme ewehe fuene cozmo marotev.
    </p>
  </div>
</span>
`;

exports[`single-connected-component render a list of connected components (no preloading) 1`] = `
<ul>
  <li>
    <div>
      id: 1
      <input
        checked={false}
        type="checkbox" />
      <p>
        Fouti hi rovumjak wuwab ojki utemun ti voblom mefup ivkiz nadahuw ci.
      </p>
    </div>
  </li>
  <li>
    <div>
      id: 2
      <input
        checked={true}
        type="checkbox" />
      <p>
        Bev rip asu rosob lo su za wirun kubvaw nadridzez cabfed temekuinu co.
      </p>
    </div>
  </li>
  <li>
    <div>
      id: 3
      <input
        checked={true}
        type="checkbox" />
      <p>
        Liuna igana pucodmek emik hapo tuczuc vokme podikru vokuse horu mum tion catke pohabe wipewen zavse osawa.
      </p>
    </div>
  </li>
  <li>
    <div>
      id: 4
      <input
        checked={false}
        type="checkbox" />
      <p>
        Mufkepgi kuretfu vucuce gardece leh nah ra gajlovzes hovwuc oca girgege zivupor ikjahwas wikesa ba ah.
      </p>
    </div>
  </li>
  <li>
    <div>
      id: 5
      <input
        checked={false}
        type="checkbox" />
      <p>
        Gem pujhi elugij mifopa bic bon ra ihaphi mit gasezuk tutfozos be iwko pulme ewehe fuene cozmo marotev.
      </p>
    </div>
  </li>
</ul>
`;

exports[`single-connected-component render a single connected component 1`] = `
<div>
  id: 1
  <input
    checked={false}
    type="checkbox" />
  <p>
    Fouti hi rovumjak wuwab ojki utemun ti voblom mefup ivkiz nadahuw ci.
  </p>
</div>
`;
